### DISCUSSION TITLE
Include all updated workspaces/packages in commit scope.

### DISCUSSION THREAD

**Discussion by simonlovesyou** [renovatebot/renovate#23991] [Suggest an Idea]
<details><summary>TL;DR:</summary>
Add configuration option `":semanticCommitScope(packages)"` similar to the existing `":semanticCommitScope(package)"` option that will include _all_ packages in the commit scope.
</details>

# Context

It's currently possible to [configure renovate to include the updated package in the commit scope](https://docs.renovatebot.com/semantic-commits/#changing-the-semantic-commit-scope) if using semantic commits in a mono repository.

But at my currrent job with our mono repository we've experienced, more often than not, that a dependency update updates more than a single package in our mono repository.

# Example

Mono repository with several packages in the `packages` directory:
```
..
├── packages/
│   ├── package-A/
│   ├── package-B/
│   └── package-C/
└── package.json
```

If a dependency is updated that all packages depend on renovate will commit

```
chore(package-C): update dependency X to Y.Z
```

But we would like it to instead commit
```
chore(package-A, package-B, package-C): update dependency X to Y.Z
```

# Justification
* By including all affected packages in the commit scope we can keep track of all individual packages and when their dependencies have been updated through tooling that makes use of the semantic commit scope.
* When a PR is created by renovate we'll get an easier overview of what packages are affected by an update.

# Suggestion

Add configuration option `":semanticCommitScope(packages)"` similar to the existing `":semanticCommitScope(package)"` option that will include _all_ packages in the commit scope.

# References
* [Renovate Docs | Semantic Commit messages # Changing the Semantic Commit scope](https://docs.renovatebot.com/semantic-commits/#changing-the-semantic-commit-scope)

**Comment by rarkins** [comment]
`":semanticCommitScope(packages)"` won't magically work like you hope unless `packages` is already the full string like `package-A, package-B, package-C`

**Reply by simonlovesyou** [reply]
Right, I see I misunderstood the documentation. To get the end result of what I thought `:semanticCommitScope(package)` would do, you would set `semanticCommitScope` to `{{parentDir}}`, right? So I guess what I'm really looking for is an added template `{{parentDirs}}` for `semanticCommitScope` (or any other descriptive name, doesn't have to be `parentDirs`)
